### DISCUSSION TITLE
User guide and usage questions (with a focus on Auto-Type)

### DISCUSSION THREAD

**Discussion by uio23** [keepassxreboot/keepassxc#11110] [Q&A] [answered]
Hello, 
I emailed KeePassXC's team a while back with some questions and issues I faced after reading their [User Guide](https://keepassxc.org/docs/KeePassXC_UserGuide.html), but never got a response. So I was hoping I could get some answers here.

1. I failed to find where exactly I could disable this option in the settings?
![image](https://github.com/user-attachments/assets/b0100b43-fcc9-4b65-8e55-02745b5a2477)
2. Would enabling Quick Unlock mean that my Apple device would be insecurely storing a temporary copy of my password database  somewhere (in order to enter it when I use Touch-ID)? 
![image](https://github.com/user-attachments/assets/ea48529b-90b6-4f39-972b-2a4756af56b4)

3. If the TOTP codes are stored in a seperate database (as per the guides recommendations), how would the Auto-Type feature be viable? Is there a way to combine values from multiple databases when performing Auto-Type? I mean:
using the Auto-Type sequence defined in the database with the username and password, and then adding in the {TOTP} field entry to that sequence, for it to fetch from another database?
![image](https://github.com/user-attachments/assets/e2563438-16a6-4e06-b592-04b346e9d438)
<br>

ISSUES:
1.
When the Auto-Type window pops up, its always out of focus. This means I have to click on it to make my field selection with the arrow keys. But when I do click on the pop-up, the focus is switched to the main KeePassXC app, meaning I have to click on the pop-up again to actual focus on it specifically. Is there any way to fix this (immediately focus on Auto-Type pop-up)?
2.
Is there a way to add/configure the Type Sequence dropdown options from the Auto-Type window? (make cmd + 1 enter: {USERNAME}{RETURN} for e.g.)
3.
Adding a * to a Window Associations seems to sometimes break Auto-Type specifically for the webpages where such a window name would match. e.g.:
Login * -> breaks Auto-Type for all webpages with "Login …” in their title.
In my case, this breaking is embodied by a new tab being opened in mozzila for "about:addons” and the password being entered into that pages search-bar.

Please note, I would like to avoid using the browser extension as a solution to any of these issues.

**Comment by droidmonkey** [comment, marked as answer]
To answer your questions...

1. It is in the View menu, documentation needs to be adjusted
2. No, the credentials are securely stored using Apple's defined methods to do so
3. If both databases are unlocked then all entries will be shown in the auto-type dialog

For your issues:

1. This is a macOS "security feature" when you are focused on a password field it won't allow other applications to steal focus
2. No and that won't be added as a feature
3. I don't understand what you are saying here

**Reply by uio23** [reply]
Hey, thank you for your response.

Regarding issue 3, its hard to pin down, but let me try clarify:
On MacOS specifically (on linux its fine), having any Window Association record under an entree's Auto-type tab, 
be that with a wild card match character * or not, and trying to perform Auto-Type of this entry on the relevant webpage (say google account entry on google's sign in page), results in the about:addons webpage being opened in firefox and the entree's password typed there.

**Reply by droidmonkey** [reply]
you may have used the same shortcut to trigger auto-type that is used to open that page in the browser.

**Reply by uio23** [reply]
But this isn't the case when I don't specify a Window Association for an entry. This is the area I'm talking about:
![image](https://github.com/user-attachments/assets/af8bef6c-743f-46da-bdb4-fae8ab766d44)
And on linux, I use the same browser.
Would you like me to provide some more specific details or open a separate issue for this?

**Reply by droidmonkey** [reply]
Window association has absolutely nothing to do with typing or anything like that. It is entirely internal to just do a match comparison for selecting entries. So there is unfortunately nothing for me to go on here because your report doesn't sound possible. What shortcut are you using for global Auto-Type on macos?

**Reply by uio23** [reply]
I use ⇧⌘V on macos.

As far as I understand window associations do have something to do with typing. They are used to select the entry for auto-type when a pages title matches the association, and possibly use a specific sequence if specified by the check-mark. That’s the impression I get from the User Guide, especially from the lightbulb section in the screenshot above. Please correct me if I’m wrong.

Otherwise, I cannot seem to pin down the Auto-type issue that occurs sometimes on macOS, I don’t use that device much now and I don’t encounter it regularly because I avoided making Window Associations in general. Apologies for the lack of insight.

Just to end it on a positive note, thank you for this incredibly useful, great application.

Edit: rephrase for conciseness; remove unnecessary explanation.

**Reply by uio23** [reply]
So is my understanding of window associations correct or not?

**Reply by droidmonkey** [reply]
You are correct, and I said that earlier, but you are incorrect in that window associations have anything to do with actual typing or anything else like that.
